firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/TransformMutation.java

```diff
@@ -121,6 +121,25 @@ public MaybeDocument applyToLocalView(
         getKey(), doc.getVersion(), newData, Document.DocumentState.LOCAL_MUTATIONS);
   }
 
+  @Override
+  public FieldMask getFieldMask() {
+    List<FieldPath> fieldMask = new ArrayList<>();
+    for (FieldTransform transform : fieldTransforms) {
+      fieldMask.add(transform.getFieldPath());
+    }
+    return FieldMask.fromCollection(fieldMask);
+  }
+
+  @Override
+  public boolean isIdempotent() {
+    for (FieldTransform transform : fieldTransforms) {
+      if (!transform.isIdempotent()) {
+        return false;
+      }
+    }
+    return true;
+  }
```

Comment:
**Update:** I see that I'm wrong (I forgot there's a separate TransformMutation for every field with a transform) and that you have a test for this. Nice work!

<s>It occurs to me that we treat the entire mutation as idempotent or not, when really it's at a field level.  E.g.:

```
doc.update({lastUpdated: FieldValue.serverTimestamp(), count: FieldValue.numericAdd(1)});
```

We only need to store a base state value for 'count', but as written, we'll end up storing base values for 'lastUpdated' and 'count'.  

This is probably non-harmful, but if we want to "fix" this, I think we'd have a getNonIdempotentFields() method instead of getFieldMask() and isIdempotent().

I don't think I care strongly though.</s>

Reply:
We may also not rebase ArrayUnion/ArrayRemove transforms momentarily.